common/Unit.cpp

```diff
@@ -187,9 +223,57 @@ void UnitBase::rememberInstance(UnitType type, UnitBase* instance)
     }
 }
 
+int UnitBase::uninit()
+{
+    TST_LOG_NAME("UnitBase", "Uninitializing unit-tests: "
+                                 << (GlobalResult == TestResult::Ok ? "SUCCESS" : "FAILED"));
+
+    if (GlobalArray)
+    {
+        // By default, this will check _setRetValue and copy _retValue to the arg.
+        // But we call it to trigger overrides and to perform cleanups.
+        int retValue = GlobalResult == TestResult::Ok ? EX_OK : EX_SOFTWARE;
+        GlobalArray[GlobalIndex]->returnValue(retValue);
+        if (retValue)
+            GlobalResult = TestResult::Failed;
+
+        for (int i = 0; GlobalArray[i] != nullptr; ++i)
```

Comment:
If GlobalArray is a list of owning pointers, would it be better to go with std::vector<std::unique_ptr<>> instead of this manual memory management? Especially with the C-style, nullptr-terminated array.

Reply:
Yes, I suppose that's what we should use. I left it as simple as possible until I got something working, but there is more room to improve, and this is one of them.